Extension/UqlFunctionCaller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Netdudes\DataSourceryBundle\Extension;
+
+use Netdudes\DataSourceryBundle\Extension\Exception\FunctionNotFoundException;
+
+class UqlFunctionCaller
+{
+    /**
+     * @var UqlExtensionContainer
+     */
+    private $uqlExtensionContainer;
+
+    /**
+     * @param UqlExtensionContainer $uqlExtensionContainer
+     */
+    public function __construct(UqlExtensionContainer $uqlExtensionContainer)
+    {
+        $this->uqlExtensionContainer = $uqlExtensionContainer;
+    }
+
+    /**
+     * @param string  $name
+     * @param array   $arguments
+     * @param Context $context
+     *
+     * @return mixed
+     * @throws Exception\FunctionNotFoundException
+     */
+    public function callFunction($name, $arguments, Context $context)
```

Comment:
For now we decided to put context on every function, no matter if it's used or not.
It kind of makes sense to have this information for every function, just like with arguments - we always have them, even if it's an empty array.

We've tried to create separate `ContextAwareUqlFunction` class, but checking `instanceof ContextAwareUqlFunction` didn't look nice.